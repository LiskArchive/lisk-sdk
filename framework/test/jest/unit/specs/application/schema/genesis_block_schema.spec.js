/*
 * Copyright Â© 2019 Lisk Foundation
 *
 * See the LICENSE file at the top-level directory of this distribution
 * for licensing information.
 *
 * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
 * no part of this software, including this file, may be copied, modified,
 * propagated, or distributed except according to the terms contained in the
 * LICENSE file.
 *
 * Removal or modification of this copyright notice is prohibited.
 */

'use strict';

const genesisBlockSchema = require('../../../../../../src/application/schema/genesis_block_schema');

describe('schema/genesis_block_schema.js', () => {
	it('genesis block schema must match to the snapshot.', () => {
		expect(genesisBlockSchema).toMatchSnapshot();
	});
});
