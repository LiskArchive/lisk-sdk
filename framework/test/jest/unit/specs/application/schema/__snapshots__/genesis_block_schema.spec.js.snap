// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`schema/genesis_block_schema.js genesis block schema must match to the snapshot. 1`] = `
Object {
  "$id": "#genesisBlock",
  "additionalProperties": false,
  "properties": Object {
    "blockSignature": Object {
      "format": "signature",
      "type": "string",
    },
    "communityIdentifier": Object {
      "type": "string",
    },
    "generatorPublicKey": Object {
      "format": "publicKey",
      "type": "string",
    },
    "height": Object {
      "minimum": 1,
      "type": "integer",
    },
    "id": Object {
      "format": "id",
      "maxLength": 20,
      "minLength": 1,
      "type": "string",
    },
    "maxHeightPreviouslyForged": Object {
      "minimum": 0,
      "type": "integer",
    },
    "maxHeightPrevoted": Object {
      "minimum": 0,
      "type": "integer",
    },
    "numberOfTransactions": Object {
      "minimum": 0,
      "type": "integer",
    },
    "payloadHash": Object {
      "format": "hex",
      "type": "string",
    },
    "payloadLength": Object {
      "minimum": 0,
      "type": "integer",
    },
    "previousBlockId": Object {
      "format": "id",
      "maxLength": 20,
      "minLength": 1,
      "type": Array [
        "null",
        "string",
      ],
    },
    "reward": Object {
      "format": "amount",
      "type": "string",
    },
    "timestamp": Object {
      "minimum": 0,
      "type": "integer",
    },
    "totalAmount": Object {
      "format": "amount",
      "type": "string",
    },
    "totalFee": Object {
      "format": "amount",
      "type": "string",
    },
    "transactions": Object {
      "items": Object {
        "additionalProperties": false,
        "properties": Object {
          "asset": Object {
            "description": "Send relevant data with transaction like delegate, vote, signature, ...",
            "type": "object",
          },
          "fee": Object {
            "format": "amount",
            "type": "string",
          },
          "id": Object {
            "format": "id",
            "maxLength": 20,
            "minLength": 1,
            "type": "string",
          },
          "senderPublicKey": Object {
            "format": "publicKey",
            "type": "string",
          },
          "signature": Object {
            "format": "signature",
            "type": "string",
          },
          "timestamp": Object {
            "min": 0,
            "type": "integer",
          },
          "type": Object {
            "type": "integer",
          },
        },
        "required": Array [
          "type",
          "timestamp",
          "senderPublicKey",
          "signature",
        ],
        "type": "object",
      },
      "type": "array",
      "uniqueItems": true,
    },
    "version": Object {
      "minimum": 0,
      "type": "integer",
    },
  },
  "required": Array [
    "communityIdentifier",
    "version",
    "totalAmount",
    "totalFee",
    "reward",
    "payloadHash",
    "timestamp",
    "numberOfTransactions",
    "payloadLength",
    "generatorPublicKey",
    "transactions",
    "blockSignature",
  ],
  "type": "object",
}
`;
